src/hlsBinaries.ts

```diff
@@ -571,7 +572,40 @@ export async function findGHCup(_context: ExtensionContext, logger: Logger, fold
   } else {
     const localGHCup = ['ghcup'].find(executableExists);
     if (!localGHCup) {
-      throw new MissingToolError('ghcup');
+      logger.info(`probing for GHCup binary`);
+      const ghcupExe = match(process.platform)
+        .with('win32', () => {
+          const ghcupPrefix = process.env.GHCUP_INSTALL_BASE_PREFIX;
+          if (ghcupPrefix) {
+            return path.join(ghcupPrefix, 'ghcup', 'bin', 'ghcup.exe');
+          } else {
+            return path.join('C:\\', 'ghcup', 'bin', 'ghcup.exe');
+          }
+        })
+        .otherwise(() => {
+          const useXDG = process.env.GHCUP_USE_XDG_DIRS;
+          if (useXDG) {
```

Comment:
Who sets this env variable? If this is specified in ~/.profile or ~/.bashrc, then we don't see it again and have basically the same issue as before. Perhaps we should look unconditionally in XDG_BIN_HOME?

Reply:
True. But if we look unconditionally, we may have ghcup binaries in multiple locations and now need to make a decision which location takes precedence. That decision could be `~/.ghcup/bin`, because that's where `ghcup upgrade` will place the binary in lack of all env vars.

But I think the current logic is also fine: we didn't find anything in PATH and now hope some of the other env vars are visible. If no env var is visible, we only check `~/.ghcup/bin`.

This is a guessing game anyway. E.g. should we assume the user wants XDG support if we find a binary in `~/.local/bin/`?

Reply:
That said: the current code requires no changing of PATH. If we look in `~/.local/bin` without XDG env var set, we need to add `~/.ghcup/bin` to PATH now, so that `ghcup upgrade` has an effect.

I like the simplicity of the current solution. It maps the ghcup code precisely.

Reply:
Ok, sounds good enough for me.